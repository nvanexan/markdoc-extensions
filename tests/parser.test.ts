import Markdoc, { ConfigType, nodes } from "@markdoc/markdoc";
import MarkdocExtendedParser, { extendedNodes } from "../src";

const config = {
  nodes: { ...nodes, ...extendedNodes },
} as ConfigType;

test("parses highlights correctly", () => {
  const ast = convert("This is a ==test==");
  expect(ast).toMatchSnapshot();

  const content = Markdoc.transform(ast, config);
  const rendered = Markdoc.renderers.html(content);
  expect(rendered).toMatchSnapshot();
});

test("parses footnotes correctly", () => {
  const ast = convert(`
  This is a paragraph with a footnote.[^1]
  
  [^1]: this is the footnote
  `);
  expect(ast).toMatchSnapshot();

  const content = Markdoc.transform(ast, config);
  const rendered = Markdoc.renderers.html(content);
  expect(rendered).toMatchSnapshot();
  expect(rendered).toContain(
    '<sup class="footnote-ref"><a id="fnref1" href="#fn1">1</a></sup>'
  );
});

function convert(example: string) {
  const parser = new MarkdocExtendedParser({
    highlights: true,
    footnotes: true,
  });
  return parser.parse(example);
}
