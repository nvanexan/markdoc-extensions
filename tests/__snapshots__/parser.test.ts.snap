// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses footnotes correctly 1`] = `
Node {
  "$$mdtype": "Node",
  "annotations": [],
  "attributes": {},
  "children": [
    Node {
      "$$mdtype": "Node",
      "annotations": [],
      "attributes": {},
      "children": [
        Node {
          "$$mdtype": "Node",
          "annotations": [],
          "attributes": {},
          "children": [
            Node {
              "$$mdtype": "Node",
              "annotations": [],
              "attributes": {
                "content": "This is a paragraph with a footnote.",
              },
              "children": [],
              "errors": [],
              "inline": true,
              "lines": [
                1,
                2,
              ],
              "location": {
                "end": {
                  "character": undefined,
                  "line": 2,
                },
                "file": undefined,
                "start": {
                  "character": undefined,
                  "line": 1,
                },
              },
              "slots": {},
              "tag": undefined,
              "type": "text",
            },
            Node {
              "$$mdtype": "Node",
              "annotations": [],
              "attributes": {
                "href": "#fn1",
                "id": "fnref1",
                "label": "1",
              },
              "children": [],
              "errors": [],
              "inline": false,
              "lines": [],
              "slots": {},
              "tag": undefined,
              "type": "footnoteRef",
            },
          ],
          "errors": [],
          "inline": false,
          "lines": [
            1,
            2,
          ],
          "location": {
            "end": {
              "character": undefined,
              "line": 2,
            },
            "file": undefined,
            "start": {
              "character": undefined,
              "line": 1,
            },
          },
          "slots": {},
          "tag": undefined,
          "type": "inline",
        },
      ],
      "errors": [],
      "inline": false,
      "lines": [
        1,
        2,
      ],
      "location": {
        "end": {
          "character": undefined,
          "line": 2,
        },
        "file": undefined,
        "start": {
          "character": undefined,
          "line": 1,
        },
      },
      "slots": {},
      "tag": undefined,
      "type": "paragraph",
    },
    Node {
      "$$mdtype": "Node",
      "annotations": [],
      "attributes": {
        "class": "footnotes",
        "ordered": true,
      },
      "children": [
        Node {
          "$$mdtype": "Node",
          "annotations": [],
          "attributes": {
            "href": "#fnref1",
            "id": "fn1",
          },
          "children": [
            Node {
              "$$mdtype": "Node",
              "annotations": [],
              "attributes": {
                "content": "this is the footnote",
              },
              "children": [],
              "errors": [],
              "inline": true,
              "lines": [
                3,
                4,
              ],
              "location": {
                "end": {
                  "character": undefined,
                  "line": 4,
                },
                "file": undefined,
                "start": {
                  "character": undefined,
                  "line": 3,
                },
              },
              "slots": {},
              "tag": undefined,
              "type": "text",
            },
          ],
          "errors": [],
          "inline": false,
          "lines": [],
          "slots": {},
          "tag": undefined,
          "type": "footnoteItem",
        },
      ],
      "errors": [],
      "inline": false,
      "lines": [],
      "slots": {},
      "tag": undefined,
      "type": "list",
    },
  ],
  "errors": [],
  "inline": false,
  "lines": [],
  "slots": {},
  "tag": undefined,
  "type": "document",
}
`;

exports[`parses footnotes correctly 2`] = `"<article><p>This is a paragraph with a footnote.<sup class="footnote-ref"><a id="fnref1" href="#fn1">1</a></sup></p><ol class="footnotes"><li class="footnote-item" id="fn1">this is the footnote<a class="footnote-anchor" href="#fnref1">â†©</a></li></ol></article>"`;

exports[`parses highlights correctly 1`] = `
Node {
  "$$mdtype": "Node",
  "annotations": [],
  "attributes": {},
  "children": [
    Node {
      "$$mdtype": "Node",
      "annotations": [],
      "attributes": {},
      "children": [
        Node {
          "$$mdtype": "Node",
          "annotations": [],
          "attributes": {},
          "children": [
            Node {
              "$$mdtype": "Node",
              "annotations": [],
              "attributes": {
                "content": "This is a ",
              },
              "children": [],
              "errors": [],
              "inline": true,
              "lines": [
                0,
                1,
              ],
              "location": {
                "end": {
                  "character": undefined,
                  "line": 1,
                },
                "file": undefined,
                "start": {
                  "character": undefined,
                  "line": 0,
                },
              },
              "slots": {},
              "tag": undefined,
              "type": "text",
            },
            Node {
              "$$mdtype": "Node",
              "annotations": [],
              "attributes": {
                "content": "test",
              },
              "children": [],
              "errors": [],
              "inline": false,
              "lines": [],
              "slots": {},
              "tag": undefined,
              "type": "mark",
            },
          ],
          "errors": [],
          "inline": false,
          "lines": [
            0,
            1,
          ],
          "location": {
            "end": {
              "character": undefined,
              "line": 1,
            },
            "file": undefined,
            "start": {
              "character": undefined,
              "line": 0,
            },
          },
          "slots": {},
          "tag": undefined,
          "type": "inline",
        },
      ],
      "errors": [],
      "inline": false,
      "lines": [
        0,
        1,
      ],
      "location": {
        "end": {
          "character": undefined,
          "line": 1,
        },
        "file": undefined,
        "start": {
          "character": undefined,
          "line": 0,
        },
      },
      "slots": {},
      "tag": undefined,
      "type": "paragraph",
    },
  ],
  "errors": [],
  "inline": false,
  "lines": [],
  "slots": {},
  "tag": undefined,
  "type": "document",
}
`;

exports[`parses highlights correctly 2`] = `"<article><p>This is a <mark>test</mark></p></article>"`;
